src/JsonApi/Serializer/ItemNormalizer.php

```diff
@@ -132,7 +132,10 @@ public function supportsDenormalization($data, $type, $format = null): bool
     public function denormalize($data, $class, $format = null, array $context = [])
     {
         // Avoid issues with proxies if we populated the object
-        if (!isset($context[self::OBJECT_TO_POPULATE]) && isset($data['data']['id'])) {
+        if (!isset($context[self::OBJECT_TO_POPULATE])
+            && isset($data['data']['id'])
+            && (null === $this->resourceMetadataFactory || !$this->resourceMetadataFactory->create($class)->getAttribute('messenger', false))
```

Comment:
```
$this->resourceMetadataFactory->create($class)
```
may be null

Reply:
`(true || x)` will never evaluate x

Reply:
Not `resourceMetadataFactory` I mean the result of calling `create` can be null https://3v4l.org/18tE1 ?

Reply:
Sorry, got it

Reply:
Fixed it